hunting/search.py

```diff
@@ -0,0 +1,163 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+
+from pathlib import Path
+import click
+from detection_rules.attack import tactics_map, technique_lookup
+from .utils import load_index_file, load_all_toml
+
+
+class QueryIndex:
+    def __init__(self, base_path: Path):
+        """Initialize with the base path and load the index."""
+        self.base_path = base_path
+        self.hunting_index = load_index_file()
+        self.mitre_technique_ids = set()
+        self.reverse_tactics_map = {v: k for k, v in tactics_map.items()}
+
+    def process_mitre_filter(self, mitre_filter: tuple):
```

Comment:
Should this method be private as well? Fine either way, just seeing that it is only called internally and does not return anything.

Reply:
Good catch, updating!